projects/iio_adpd1080/src/main.c

```diff
@@ -109,17 +111,14 @@ static int32_t adpd1080pmod_32k_calib(struct adpd188_dev *adpd1080_dev)
 	if(status != 0)
 		goto gpio_finish;
 
-	struct gpio_irq_config sync_irq_config = {
-		.gpio_handler = sync_gpio,
-		.mode = GPIO_GROUP_POSITIVE_EDGE
-	};
 	struct no_os_callback_desc sync_gpio_cb = {
-		.legacy_callback = adpd1080_sync_gpio_cb,
+		.callback = adpd1080_sync_gpio_cb,
 		.ctx = &sync_gpio_pulse_no,
-		.legacy_config = &sync_irq_config
+		.event = NO_OS_EVT_GPIO,
+		.handle = sync_gpio,
```

Comment:
On the other (Maxim and STM) platforms, this `handle` field stores a "HAL descriptor" (e.g. in the Maxim's case that's a pointer to a specific GPIO port's set of registers) instead of a no-OS gpio descriptor. That's also what the comment for this field in `no_os_irq.h` says. It would be nice to keep the "non-extra" void* parameters meanings the same across all platforms, but if this is not possible on ADuCM, your current version is fine by me.

Reply:
Storing the no-os GPIO descriptor seems to me that it makes the application more platform independent. Should we change or keep like this?